RFS/src/main/java/org/opensearch/migrations/bulkload/common/DocumentReindexer.java

```diff
@@ -76,8 +81,8 @@ Mono<Void> sendBulkRequest(UUID batchId, List<BulkDocSection> docsBatch, String
                 .log())
             // Prevent the error from stopping the entire stream, retries occurring within sendBulkRequest
             .onErrorResume(e -> Mono.empty())
-            .then() // Discard the response object
-            .subscribeOn(scheduler);
+            .then(Mono.just(new IndexAndShardCursor(indexName, shardNumber, lastDoc.getLuceneSegId(), lastDoc.getLuceneDocId()))
```

Comment:
I've been having a hard time understanding why shardNumber is required here.  Now I understand - because indexName really shouldn't be required either.  This call path would be easier to understand and to maintain if it were to use an IndexAndShardCursor factory that took the lucene/docId in - or if those process global things weren't included in the tracking objects.

Since indexName/shardName are global to the reindexer (it never does more than one shard, right?) - these things shouldn't be here.  As it is now, there's a leaky abstraction.

Reply:
Shard Number and Index Name removed from here